### ISSUE TITLE
Revision not found on windows-latest agent

### ISSUE THREAD

**Issue by matejskubic** [PaulHatch/semantic-version#6]
Action can't find existing tag.

Returns next log:
```
Run paulhatch/semantic-version@v3.1.1
  with:
    branch: master
    tag_prefix: v
    major_pattern: (MAJOR)
    minor_pattern: (MINOR)
    format: ${major}.${minor}.${patch}
The command 'git describe --tags --abbrev=0 --match=v*[0-9.] origin/master~1' failed: Error: The process 'C:\Program Files\Git\bin\git.exe' failed with exit code 128
stderr: fatal: No names found, cannot describe anything.

Warning: No tags are present for this repository. If this is unexpected, check to ensure that tags have been pulled from the remote.
Version is 0.0.1+0
To create a release for this version, go to https://github.com/_org_/_repo_/releases/new?tag=v0.0.1&target=master
```

Tag & release v0.0.1 exists. Running `git describe --tags --abbrev=0 --match=v*[0-9.]` locally returns `v0.0.1`.

I'm getting error with additional parameter **`origin/master~1`**.

Sample pipeline.yml:
```yaml
name: CI
on:
  push:
  pull_request:
  workflow_dispatch:

jobs:
  build:
    runs-on: windows-latest
    steps:
      - uses: actions/checkout@v2
      - uses: paulhatch/semantic-version@v3.1.1
        with:
          branch: main
```

**Comment by PaulHatch** [comment]
I've been meaning to add this to the documentation, when using newer versions of checkout, tags will not be pulled unless you specify the `fetch-depth` parameter, specify 0 to pull all history:

```
      - name: Checkout
        uses: actions/checkout@v2
        with:
          fetch-depth: 0
```

**Comment by matejskubic** [comment]
Thank you from checkout@v2 hint.

I've updated action to use namespace (`taskOne`).

Now I'm getting next error:

```
The command 'git describe --tags --abbrev=0 --match=v*[0-9.]-taskOne origin/main~1' failed: Error: The process 'C:\Program Files\Git\bin\git.exe' failed with exit code 128
stderr: fatal: No names found, cannot describe anything.

Warning: No tags are present for this repository. If this is unexpected, check to ensure that tags have been pulled from the remote.
Version is 0.0.1+0
```

If tag exists (`v0.2.4-taskOne`) then the error is:
```
The command 'git describe --tags --abbrev=0 --match=v*[0-9.]-taskOne origin/main~1' failed: Error: The process 'C:\Program Files\Git\bin\git.exe' failed with exit code 128
stderr: fatal: No tags can describe '48a36795b2a1dd6e6239f1fef48a83670130eb37'.
Try --always, or create some tags.

```
Updated action
```yaml
name: CI
on:
  push:
  pull_request:
  workflow_dispatch:
jobs:
  build:
    runs-on: windows-latest
    steps:
      - uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - uses: PaulHatch/semantic-version@v3.1.1
        with:
          branch: main
          namespace: taskOne

```

**Comment by PaulHatch** [comment]
Thanks for sharing this. So the issue is that you have a repository which is tagged on the commit you're trying to version:

- HEAD implied v0.0.1 but tagged v0.2.4
- HEAD~1 implied v0.0.1

Semantic version's goal is to give you a version indicating what the next version _will be if released_ in order to solve the chicken an egg problem you have when you're making a build but you don't know yet if it will pass whatever release process you have, therefore in terms of tags it is looking only at recent tags starting at the previous commit and uses that to calculate the next version number.

Normally this would be fine because if this commit had been tagged with what this action was expecting the tag to be it would be correct, the problem here is that the next version number skipped ahead from the expected number and this isn't taken into account. The following commit would indicate 0.2.5, being the next patch increment after 0.2.4.

I am adding logic to check if the current commit is tagged, and if so override using the value from the current commit so that this scenario will work in the future.

**Comment by PaulHatch** [comment]
Fixed in v3.1.2

**Comment by matejskubic** [comment]
> I am adding logic to check if the current commit is tagged, and if so override using the value from the current commit so that this scenario will work in the future.

Thank you for great explanation and effort in deep issue analysis.
